### DISCUSSION TITLE
how to make apps for cosmos c#

### DISCUSSION THREAD

**Discussion by EndermanPC** [CosmosOS/Cosmos#1999] [General]
i finished the work with cosmos but i cant find any way to make an app on cosmos
I'm new to cosmos and I'm not very good at it
help me
thanks in advance

**Comment by Andreas010** [comment]
Please further describe your issue, e.g. What do you mean by "App".

If you want something like the Windows cmd or others, where you can type an app's name and make it execute, you gotta do that on your own in C#, e.g. with a Console.ReadLine() and then somehow figuring out, what to run with what arguments

**Reply by sajjadarashhh** [reply]
Make Gui Applications somethings like windows form or wpf
we can do this!?

**Reply by AsertCreator** [reply]
> Make Gui Applications somethings like windows form or wpf we can do this!?

well you can, but its very hard. you need to implement everything yourself. all controls, forms/windows, another things, all of them you need to implement yourself. its very hard, isnt it?

**Comment by EndermanPC** [comment]
hmm, i just want to create a simple application like cmd on windows, i want us to enter a deep command and it will open, i don't know how to activate an application in cosmos

**Reply by partisani** [reply]
> hmm, i just want to create a simple application like cmd on windows, i want us to enter a deep command and it will open, i don't know how to activate an application in cosmos

Simple, I Create A new C# File (**CommandLineApp.cs**) So I Write The Cmd, Now In The Main File (**Kernel.cs**) I Call The **CommandLineApp.cs**. I Said That Will Be Simple!

**Comment by Andrej123456789** [comment]
@EndermanPC I think you think about third party developer which create executable for your OS developed by Cosmos Kit

**Reply by partisani** [reply]
> @EndermanPC I think you think about third party developer which create executable for your OS developed by Cosmos Kit

In This Side Is Better Create A New Programming Language But You Will Need To Have A VFS

**Comment by Andrej123456789** [comment]
Your question is intersting

**Comment by sajjadarashhh** [comment]
this can be powered and very use full to take ability to run .net application in cosmos os as container to create mini and cheep Computer in Companies  for their Projects

**Comment by kuskebabi** [comment]
i think you need to make your own programming language for your os

**Reply by partisani** [reply]
yea, but thats a hard one, so try with caution. ( I dont use Cosmos anymore)

**Comment by AsertCreator** [comment]
use executables. you need to parse it, link something, and jump to it.

**Reply by Andrej123456789** [reply]
Agree

**Comment by IDONTCAREE44** [comment]
Is there is any way to create a full browser? maybe a file system GUI ?
IN A very very clear simple way? maybe how to add a framework Application form in a cosmos project?

**Reply by ascpixi** [reply]
You'd have to implement that yourself - OS development is pretty challenging, even with Cosmos, which already simplifies it a lot and abstracts the majority of the architectural intrinsics behind the HAL (and provides you with lots of drivers - normally, you'd have to write them yourself). AFAIK, there's no open-source browser that can work with the runtime library Cosmos provides.

**Comment by ytcyberslotp90** [comment]
> i finished the work with cosmos but i cant find any way to make an app on cosmos I'm new to cosmos and I'm not very good at it help me thanks in advance

Umm... I know what do you mean, you want your OS to run Its apps, right ??
I made it, you have to make executables of your own, from scratch.

Like I made a executable which is written in **ASSEMBLY** and I made my own **Bytecode** pattern and Compiled it with my own compiler and made an interpreter for that too.

It seems very hard, but it isn't.
